Refactored the load script.

GabrielNsiah opened the pull request:
Refactored the load script for ticket #6. Includes:

- More logging.
- More error handling.
- No more -1 as a return value.
- Various code refactored.
- Verification for `.env` values.

bentrzcinski commented on pipeline/load.py:
```diff
@@ -44,65 +55,38 @@ def get_connection() -> extensions.connection:
 def get_id_from_release_info(release_name: str, release_date: datetime,
-                             artist_id: int, type_id: int, cursor: extensions.cursor):
+                             artist_id: int, type_id: int,
+                             cursor: extensions.cursor) -> Optional[int]:
```
Last week we got this feedback:
"Your return types can be dict | None, you don’t need to import optional and do Optional[dict]"
I think the same applies here

bentrzcinski replied:
And all the other instances in different functions

bentrzcinski commented on pipeline/load.py:
```diff
@@ -257,34 +241,73 @@ def insert_release_genres(release_id: int, genre_id: int, genre_name: str,
+def insert_sale_data(sale_price: float, sale_date: datetime,
+                     country_name: str, release_id: int, cursor) -> None:
+    """"
+    Inserts the sale data relating to a release into the database
```
```suggestion
    Inserts the sale data relating to a release into the database.
```

bentrzcinski commented on pipeline/load.py:
```diff
@@ -24,7 +24,18 @@ def config_log() -> None:
 
 
-def get_connection() -> extensions.connection:
+def validate_env_vars() -> None:
+    """
+    Raises an error if any environment variables are missing
+    """
+    required_vars = ["DB_HOST", "DB_PORT", "DB_USER", "DB_PASSWORD", "DB_NAME"]
+    missing_vars = [var for var in required_vars if var not in environ]
+    if missing_vars:
+        raise EnvironmentError(
+            f"Missing required environment variables: {missing_vars}")
```
This is really nice

bentrzcinski reviewed (changes requested):
Really nice. I especially like all the error handling.

GabrielNsiah commented:
Implemented the above changes, and also:
- Changed the formatting in `transform.py`, so that all dates are in the format `YYYY-MM-DD` for consistency.

bentrzcinski reviewed (approved):
great stuff

emily-curtis reviewed (approved):
looks good!

ebradley12 commented on pipeline/load.py:
```diff
@@ -44,65 +54,38 @@ def get_connection() -> extensions.connection:
 
-def get_id_from_country(country_name: str, cursor: extensions.cursor) -> int:
+def get_id_from_table(search_value: str, table_name: str,
+                      cursor: extensions.cursor) -> int | None:
```
Really nice - makes the code more DRY :)

ebradley12 commented on pipeline/load.py:
```diff
@@ -139,30 +123,30 @@ def insert_country(country_name: str, cursor: extensions.cursor) -> None:
-def insert_genres(genre_name: str, cursor: extensions.cursor) -> None:
+def insert_genres(genre_name: str, cursor: extensions.cursor) -> int | None:
     """
     Inserts genres into the database they don't already exist.
```
Can you add 'if'

ebradley12 commented on pipeline/transform.py:
```diff
@@ -57,7 +57,7 @@ def convert_from_unix_to_datetime(unix: str) -> datetime:
+def convert_written_date_format(date_str: str) -> str:
     """
     Takes dates in the "DD B YYYY" format and
     converts them to 'DD-MM-YYYY'.
```
Update docstrings to YYYY-MM-DD

GabrielNsiah commented:
Inserted the above changes.
